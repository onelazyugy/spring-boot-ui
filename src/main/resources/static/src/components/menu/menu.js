import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';

export class Menu extends Component {

    constructor(props, context) {
        super(props, context);
    }

    selectMenu = (event) => {
        let parent = document.getElementById('menu');
        for (let i = 0; i < parent.children.length; i++) {
            let anchor = parent.children[i].children[0];
            anchor.classList.remove('active');
        }
        document.getElementById(event.target.id).classList.add('active');

        let id = event.target.id;
        let newRoute = '/';
        if (id === 'home') {
            newRoute += 'home';
        } else {
            newRoute += 'home/' + event.target.id;
        }
        this.props.history.push(newRoute);
    }

    render() {
        return (
            <div className="sidenav scrollable">
                <ul className="nav" id="menu">
                    <li><a onClick={this.selectMenu} className="active" id="home"><i className="icon_home" />Home</a></li>
                    <li><a onClick={this.selectMenu} id="components"><i className="icon_dashboard" />Components</a></li>
                </ul>
            </div>
        );
    }
}

Menu.propTypes = {
    history: PropTypes.object
};

export default withRouter(Menu);
