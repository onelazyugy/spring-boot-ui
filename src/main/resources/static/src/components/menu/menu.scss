.navigation-wrapper .sidenav {
    z-index: 0;

    ul.nav {
        margin-top: 130px;
        margin-bottom: 10px;

        li {
            cursor: pointer;
        }
    }
}
