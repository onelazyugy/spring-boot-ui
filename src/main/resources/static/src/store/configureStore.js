import { createStore, compose, applyMiddleware } from 'redux';
import reduxImmutableStateInvariant from 'redux-immutable-state-invariant';
import thunk from 'redux-thunk';
import { createBrowserHistory } from 'history';
import { connectRouter, routerMiddleware } from 'connected-react-router';
import rootReducer from '../reducers';

const history = createBrowserHistory();
const routing = routerMiddleware(history);

function configureStoreProd(initialState) {
    const middlewares = [
        routing,
        thunk
    ];

    return createStore(
        connectRouter(history)(rootReducer),
        initialState,
        compose(
            applyMiddleware(...middlewares)
        )
    );
}

function configureStoreDev(initialState) {
    const middlewares = [
        reduxImmutableStateInvariant(),
        routing,
        thunk
    ];

    const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose; // add support for Redux dev tools
    const store = createStore(
        connectRouter(history)(rootReducer),
        initialState,
        composeEnhancers(
            applyMiddleware(...middlewares)
        )
    );

    if (module.hot) {
        // Enable Webpack hot module replacement for reducers
        module.hot.accept('../reducers', () => {
            const nextReducer = require('../reducers').default; // eslint-disable-line global-require
            store.replaceReducer(nextReducer);
        });
    }

    return store;
}

const configureStore = process.env.NODE_ENV === 'production' ? configureStoreProd : configureStoreDev;

export default configureStore;
