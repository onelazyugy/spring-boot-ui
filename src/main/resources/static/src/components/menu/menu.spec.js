import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'store/configureStore';
import { Menu } from './menu';

describe('<Menu />', () => {
    let store, wrapper;

    beforeAll(() => {
        store = configureStore();
        wrapper = shallow(<Menu />, { context: { store: store } });
    });

    it('should match its empty snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

});
