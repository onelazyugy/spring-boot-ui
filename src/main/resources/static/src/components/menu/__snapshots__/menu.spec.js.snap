// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should match its empty snapshot 1`] = `
<div
  className="sidenav scrollable"
>
  <ul
    className="nav"
    id="menu"
  >
    <li>
      <a
        className="active"
        id="home"
        onClick={[Function]}
      >
        <i
          className="icon_home"
        />
        Home
      </a>
    </li>
    <li>
      <a
        id="components"
        onClick={[Function]}
      >
        <i
          className="icon_dashboard"
        />
        Components
      </a>
    </li>
  </ul>
</div>
`;
